### PR TITLE
SED-3546 adding cache control filter

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public void init(FilterConfig filterConfig) {}
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        if (servletRequest instanceof Request) {
+            Request request = (Request) servletRequest;
+            HttpServletResponse httpResponse = (HttpServletResponse) response;
```
Comment (jeromecomte) [inline step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:24]: I assume it is safe but we could add an explicit type check if we want to be 100% safe

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+            // Check if the request is for the index HTML file which is also implicit for the root URI context (welcome file)
+            if (request.getRequestURI().endsWith("index.html") || request.getPathInContext().equals("/")) {
+                // Set Cache-Control header to never cache HTML files
```
Comment (jeromecomte) [inline step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:28]: The comment not exactly matches the if clause which applies only to index.html.

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        if (servletRequest instanceof Request) {
+            Request request = (Request) servletRequest;
+            HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+            // Check if the request is for the index HTML file which is also implicit for the root URI context (welcome file)
+            if (request.getRequestURI().endsWith("index.html") || request.getPathInContext().equals("/")) {
```
Comment (jeromecomte) [inline step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:27]: What is the reason of the check request.getPathInContext().equals("/")? Just for my understanding

**Review by jeromecomte — COMMENTED** [review]
Having some remarks and one question

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java (hunk 3 above)
Comment (david-stephan) [reply step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:27]: @jeromecomte well this is what's in comment but apparently not very clear. There are actually no request to index.html directly just keeping it to be safe, the important request is to "http(s)://hostname:port" for which the path in context is "/". Jetty then automatically return the welcome page (i.e. index.html).

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java (hunk 2 above)
Comment (david-stephan) [reply step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:28]: @jeromecomte see my answer to next comment

File: step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java (hunk 1 above)
Comment (david-stephan) [reply step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java:24]: It is safe, but I'll add a check for the form.

**Review by jeromecomte — APPROVED** [review]

**Pull request merged**